drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -73,4 +80,53 @@ def factory_name
   def schema
     raise 'Implement in sub-class'
   end
+
+  def process_custom_field(field, value)
+    record = @processor.send(factory_name)
+    return false unless record.respond_to?(:custom_data_elements)
+
+    cded = Hmis::Hud::CustomDataElementDefinition.for_type(record.class.name).where(key: field).first
```

Comment:
There are a lot of places in the HMIS driver where we user `class.name` instead of `class.sti_name`.. it seems like with the current setup [they are equivalent](https://apidock.com/rails/ActiveRecord/Inheritance/ClassMethods/sti_name), but maybe we should switch to using sti_name in case we ever want to set `store_full_sti_class = false` so that the type column gets stored as the demodulized version. Is that your reasoning for suggesting it?
If we change it in one place, we should change it everywhere.

I made the find_by change.

Reply:
Hmm ok. How do I know if something is coming from a call to `sti_name` in the background?
We usually set these polymorphic values by just doing something like `owner: client`, so how do I know if that's going to use `client.class.sti_name` or `client.class.name` to set the `owner_type` field? sorry, I'm not finding good docs about this.

Reply:
Hmm, I'm not sure where to look for docs, but anytime rails is deciding what to put into an STI or polymorphic column that stores the class name, it'll use `sti_name`.  If we do it manually, we may have put `name` in there....

Reply:
Yes, when we do it manually we are using `name`. Sounds like we should update everything to use `sti_name`. I'm going to handle this in a separate PR.

Reply:
Sounds like a solid plan.  Thanks!